### PR TITLE
pkg/sdk: Add label selector for sdk.Watch()

### PR DESCRIPTION
**Pull request by CSdread**
pkg/sdk: adding label selector to sdk.Watch()

This is being made because as of right now there is no way to watch a specific
sub-set of a resource based on labels. This makes implementing finalizer 
controllers difficult.

### CODE REVIEW

**Comment by theishshah** [comment]
Can we document the update to the Watch interface in the user guide [here](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#watch-the-memcached-cr)

**Comment by CSdread** [comment]
@theishshah added some documentation and propose slight format change since we now have more than one option for watch.

File: doc/user-guide.md [hunk 1]
```diff
@@ -65,11 +65,22 @@ func main() {
...
-**Note:** The number of concurrent informer workers can be configured with an additional Watch option. The default value is 1 if an argument is not given.
+#### Options
+**Worker Count**
+The number of concurrent informer workers can be configured with an additional Watch option. The default value is 1 if an argument is not given.
 ```Go
 sdk.Watch("cache.example.com/v1alpha1", "Memcached", "default", 5, sdk.WithNumWorkers(n))
 ```
 
+**Label Selector**
+Label selectors allow the watch to filter resources by kubernetes labels. It can be sepecified using the standard kubernetes label selector format:
```
Comment (hasbro17) [inline doc/user-guide.md]: *specified

**Comment by hasbro17** [comment]
@CSdread Were you able to test this out to watch for CRs with a particular label? 
What about specifying multiple label selectors `app=test,foo=bar`? Can you verify that works too.

**Comment by CSdread** [comment]
@hasbro17 it takes the k8s label selector format as its argument. It should handle what you would build in an operator to do the label selection.

**Comment by CSdread** [comment]
I will verify it tonight

**Comment by hasbro17** [comment]
LGTM
@fanminshi PTAL

**Comment by fanminshi** [comment]
lgtm

**Pull request merged**